### PR TITLE
Complete tasks for open source

### PR DESCRIPTION
**Pull request by JoyceXinyueWang**
In this pull request:

* Add badges to readme
* Add tests to achieve 90%+ coverage
* Modify azure pipeline yaml file to include tests on notebooks and build artifact for release pipeline
* Add docs for readthedocs - currently I have difficulty build the docs with blocklib since branch open-source-docs is not found in readthedocs

### CODE REVIEW

File: docs/tutorial/index.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Tutorials
+=========
+
```
Comment (wilko77) [inline docs/tutorial/index.rst]: this needs some text here. Have a look at what we did in clkhash.

File: docs/blocking-schema.rst [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+Blocking Schema
+===============
+Each blocking method has its own configuration and parameters to tune with. To make our API as generic
+as possible, we designed the blocking schema to specify the configuration of the blocking method including
+features to use in generating blocks and hyperparameters etc.
+
+Currently we support two blocking methods:
+
+* "`p-sig`": Probability signature
```
Comment (wilko77) [inline docs/blocking-schema.rst]: Probabilistic signatures

File: docs/blocking-schema.rst [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+.. _blocking-schema/lambda-fold:
+
+LSH based :math:`\lambda`-fold Configuration
+~~~~~~~~~~~~~~~~~~~~~
+===================== ============= ==========================
+attribute             type          description
+===================== ============= ==========================
+blocking-features     list[integer] specify which features to used in blocks generation
+Lambda                integer       denotes the degree of redundancy - :math:`H^i`, :math:`i=1,2,...`, :math:`\Lambda` where each :math:`H^i` represents one independent blocking group
```
Comment (wilko77) [inline docs/blocking-schema.rst]: I've never noticed that before, but why are some parameters upper case and others don't?

File: tests/test_simmeasure.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    assert score > 0
+
+
+def test_dicesim():
+    """Test Dice similarity measure."""
+    config = dict(ngram_len=2, ngram_padding=1, padding_start_char='a', padding_end_char='z')
+    sim = DiceSim(config)
+
+    score = sim.sim('Joyce wears a T-shrt today', 'Jayce wear an T-shrt today', cache=True)
+    assert score > 0.8
```
Comment (wilko77) [inline tests/test_simmeasure.py]: these tests are fairly minimal. I can think of many implementations for EditSim and DiceSim that would pass those tests but do not compute the actual Dice and Edit similarities.

File: blocklib/__init__.py [hunk 5]
```diff
@@ -7,3 +8,8 @@
 from .candidate_blocks_generator import generate_candidate_blocks
 from .encoding import generate_bloom_filter, flip_bloom_filter
 from .evaluation import assess_blocks_2party
+
+try:
+    __version__ = pkg_resources.get_distribution('anonlinkclient').version
```
[the remark this answers is not in the document]
Comment (JoyceXinyueWang) [reply blocklib/__init__.py]: oops

File: docs/blocking-schema.rst (hunk 3 above)
Comment (JoyceXinyueWang) [reply docs/blocking-schema.rst]: To follow the paper where it uses upper case for some parameters and lowercases for others

**Pull request merged**